### ISSUE TITLE
Training Uni-Sign on a small custom LSQ dataset (no WLASL) + missing pre_training.py

### ISSUE THREAD

**Issue by ga3ll3y** [ZechengLi19/Uni-Sign#33]
Hello,

I would like to train Uni-Sign on LSQ (Quebec Sign Language) dataset, without using WLASL, since the languages and signs are very different.

However, I encountered two issues:

1) Missing pre-training script  
In the repository, train_stage1.sh and train_stage2.sh both call pre_training.py, but this file does not seem to be included.  
Is the pre-training code available somewhere, or was it not released?

2) Training on a small dataset  
My dataset contains:
- ~525 classes
- ~950 training videos  
(many classes have only 1–2 samples)

When training directly with:
    fine_tuning.py --task ISLR

the model quickly collapses and predicts only a few frequent classes, with near-zero accuracy.

My questions are:

- Do you recommend using a pretrained checkpoint (e.g., WLASL / CSL-News) even if the target sign language is different (LSQ)?
- Is there a recommended strategy for small datasets (few-shot / low-resource)?
- Would you suggest:
  - pose-only training instead of RGB?
  - freezing the backbone and training only the classifier?
  - or a different training setup?

More generally, what is the recommended pipeline to adapt Uni-Sign to a new isolated sign language dataset with very limited data?

Thank you for your help!

**Comment by ZechengLi19** [comment]
@ga3ll3y You can find the pretrain code at: [here](https://github.com/ZechengLi19/Uni-Sign/blob/eed438bcb49e30405cd6ccdfcccca330c134e830/README.md?plain=1#L69).
 
For your low-resource training setting, my suggestion is:
 
1. Initialize the model with the released pretrained weights.
2. Re-initialize the weights of the LLM classification head.
3. Only train the classification head for a few epochs first.
4. Then proceed with full-parameter training.
 
This pipeline is stable and works well for low-resource scenarios.

**Comment by ga3ll3y** [comment]
Thanks for your answer.

I followed the suggested low-resource pipeline on my LSQ dataset:

1. LSQ-only pose pretraining
2. LSQ RGB+pose pretraining
3. ISLR fine-tuning
4. Then I tried the head-warmup strategy:
   - re-initialize `mt5_model.lm_head`
   - freeze all parameters except `lm_head`
   - train the head for a few epochs
   - then run full-parameter fine-tuning

The head-warmup strategy improved the dev result:
- RGB+pose LSQ without head warmup: max dev PI accuracy = 6.56%
- RGB+pose LSQ with head warmup: max dev PI accuracy = 9.84%

However, the test performance is still very low. The previous RGB+pose LSQ model achieved only:
- 6 correct predictions out of 517 test videos
- around 1.16% test accuracy

My LSQ dataset is very low-resource:
- train: 950 videos
- dev: 61 videos
- test: 517 videos
- total classes: 525

So the model has only around 1-2 training samples per class on average. The predictions often collapse to a small set of glosses.

Do you think Uni-Sign’s generative mT5 formulation is suitable for this kind of isolated-sign low-resource setting?

Thanks,

**Comment by ZechengLi19** [comment]
@ga3ll3y Given that your data scenario is extremely low-resource, additional refined few-shot settings will be required to improve overall model performance. For general optimization, you can start from the following two aspects:

1. Data-level optimization: You can expand your existing video dataset or apply video augmentation to increase the diversity of skeleton point samples, which effectively alleviates overfitting caused by insufficient data.
2. Error analysis: It is also necessary to conduct systematic failed case analysis to find which category is most frequently mispredicted, helping you locate problems in data or model design.

Finally, please double-check that the annotation format and coordinate definition of your skeleton point data are fully consistent with Uni-Sign to avoid compatibility issues affecting performance.
